File: app/src/main/java/hypr/a255bits/com/hypr/Generator/Control.java

package hypr.a255bits.com.hypr.Generator;

import java.util.List;
import com.squareup.moshi.Json;

public class Control {

    @Json(name = "type")
    private String type;
    @Json(name = "direction")
    private List<Double> direction = null;
    @Json(name = "name")
    private String name;

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public List<Double> getDirection() {
        return direction;
    }

    public void setDirection(List<Double> direction) {
        this.direction = direction;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

}

File: app/src/main/java/hypr/a255bits/com/hypr/Generator/Control.kt
package hypr.a255bits.com.hypr.Generator

import android.os.Parcelable
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

@Parcelize
class Control : Parcelable {
    @Json(name = "type")
    var type: String? = null

    @Json(name = "name")
    var name: String? = null
}
